core/src/main/kotlin/knote/script/NotebookConfiguration.kt

```diff
@@ -64,47 +73,52 @@ class NotebookConfiguration : ScriptCompilationConfiguration({
                     )
                     return@onAnnotations ResultWithDiagnostics.Failure(reports)
                 }
-            } ?: return@onAnnotations context.compilationConfiguration.asSuccess(reports)
+            }
 
             // TODO: list pages, generate constants and include
             val pagesDir = rootDir.resolve("${id}_pages")
             val pages = pagesDir.listFiles { file -> file.isFile && file.name.endsWith(".page.kts") } ?: run {
                 reports += ScriptDiagnostic("no files found in $pagesDir", ScriptDiagnostic.Severity.ERROR)
                 arrayOf<File>()
-//                return@onAnnotations ResultWithDiagnostics.Failure(reports)
+//                return@onAnnotations ResultWithDiagnostics.Failure(reportsObject)
             }
 
             val generatedSrc = rootDir.resolve("build").resolve(".knote").resolve(id)
             val generatedFiles = PageMarker.generate(generatedSrc, pages, fileName = id.capitalize())
 
-            reports += ScriptDiagnostic("generated: ${generatedFiles.map { it.relativeTo(rootDir) }}", ScriptDiagnostic.Severity.INFO)
+            reports += ScriptDiagnostic(
+                "generated: ${generatedFiles.map { it.relativeTo(rootDir) }}",
+                ScriptDiagnostic.Severity.INFO
+            )
 
-            val compilationBuilder = ScriptCompilationConfiguration(context.compilationConfiguration) {
+            return@onAnnotations ScriptCompilationConfiguration(context.compilationConfiguration) {
+                ide.acceptedLocations.append(ScriptAcceptedLocation.Project)
                 if (generatedFiles.isNotEmpty()) {
                     importScripts.append(generatedFiles.map { it.toScriptSource() })
                 }
-            }
-
-            val importAnnotations = annotations.filterIsInstance(Import::class.java)
-            reports += ScriptDiagnostic("importAnnotations: $importAnnotations", ScriptDiagnostic.Severity.DEBUG)
-
-            val sources = importAnnotations.map {
-                rootDir.resolve("include").resolve(it.source)
-            }.distinct()
 
-            return@onAnnotations compilationBuilder.apply {
-                if (sources.isNotEmpty()) {
-                    importScripts.append(sources.map { it.toScriptSource() })
-                    reports += ScriptDiagnostic(
-                        "importScripts += ${sources.map { it.relativeTo(rootDir) }}",
-                        ScriptDiagnostic.Severity.INFO
-                    )
+                if(annotations != null) {
+                    val importAnnotations = annotations.filterIsInstance(Import::class.java)
+                    reports += ScriptDiagnostic("importAnnotations: $importAnnotations", ScriptDiagnostic.Severity.DEBUG)
+
+                    val sources = importAnnotations.map {
+                        rootDir.resolve("include").resolve(it.source)
```

Comment:
i like a resolve call for each file level, no need for string interpolation either
and i see there is still importAnnotations there.. thoose have to be wiped since they break dependencies